### ISSUE TITLE
Object to String and String to Object Hashmap

### ISSUE THREAD

**Issue by aaron-ng** [TomPeng19971208/CS4500_A1_Part2#7]
There is a currently an issue with your definition where when you build an HashMap you can insert a key value pair as (String, Object) or (Object, String) which goes against the assignment spec of only (String, String) or (Object, Object).

You could add a subclass that only allows you to add a (String, String) value pair so that we can guarantee that all values in the HashMap is a String or an Object. 

I know this is a pain sorry :(

**Comment by thomas-vaiarella** [comment]
In the spec, I see: "For Map, you should support String to String and String to Object **(at least)**"
In my opinion, I think this group's implementation adheres to this requirement.

**Comment by TomPeng19971208** [comment]
Map allows Object -> Object mapping. If you have a Map<String, Integer>,  and the given key value pair is not valid, in your implementation, the program should error. Basically, this is an interface, validation should be done in the implementation. Making sure all entry-value pairs in the map have the same type is part of the implementation.

**Comment by BlueSpud** [comment]
IMO there are a lot of benefits for also including what Aaron is saying, so long as you don't use inheritance. This lets the compiler enforce the type invariant for you and also communicates much more clearly to readers of the code that a map is from a String->Object rather than a Object->Object and then getting an error thrown when not using a string

**Comment by TomPeng19971208** [comment]
There is an infinite number of Object Type,. I wouldn't imagine adding a subclass for map/queue/array every time a new class is defined.

**Comment by BlueSpud** [comment]
Per https://piazza.com/class/k51bluky59n2jr?cid=331 it seems to be the instructors opinion that you should have these

**Comment by xiaju** [comment]
Agreed with Aaron that since we need to explicitly support String->String, a subclass that only allows a Strings would be a good idea. That said I think simple method overloading might suffice as well. Regardless, I think we need a way in the interface to know whether a user is trying to do String->String or the more generic Object->Object.

**Comment by TomPeng19971208** [comment]
I will bring this up to the professor tomorrow in class.

**Comment by TomPeng19971208** [comment]
U have to make other teams on board with, since this might change their implementation, and I don't think this is a good practice.
